src/session.rs

```diff
@@ -131,6 +135,43 @@ pub fn create_batch(queries: Vec<&PreparedStatementWrapper>) -> BatchWrapper {
     BatchWrapper { inner: batch }
 }
 
+fn apply_options(
+    mut query: PreparedStatement,
+    options: &QueryOptionsWrapper,
+) -> napi::Result<PreparedStatement> {
+    if let Some(o) = options.consistency {
+        query.set_consistency(
+            Consistency::try_from(o)
+                .map_err(|_| js_error(format!("Unknown consistency value: {o}")))?,
+        );
+    }
+    if let Some(o) = options.is_idempotent {
+        query.set_is_idempotent(o);
+    }
+    // TODO: Update it and check all edge-cases:
+    // https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/pull/92#discussion_r1864461799
+    // Currently there is no support for paging, so there is no need for this option
+    /* if let Some(o) = options.fetch_size {
+        if o.is_negative() {
+            return Err(js_error("fetch size cannot be negative"));
+        }
+        query.set_page_size(o);
+    } */
+    if let Some(o) = &options.timestamp {
+        query.set_timestamp(Some(bigint_to_i64(
+            o.clone(),
+            "Timestamp cannot overflow i64",
+        )?));
+    }
+    // TODO: Update it to allow collection of information from traced query
+    // Currently it's just passing the value, but not able to access any tracing information
+    if let Some(o) = options.trace_query {
+        query.set_tracing(o);
+    }
```

Comment:
:question: IIUC, we will be able to turn on tracing, but we won't be able to read the returned tracing data?